docs/tutorials/customSessionKeys/customSessionStorage.md

```diff
@@ -0,0 +1,268 @@
+
+```typescript
+import * as fs from "fs";
+import { Wallet, Signer } from "ethers";
+import { ISessionStorage, SessionLeafNode, SessionSearchParam, SessionStatus } from "@biconomy/modules/dist/src/interfaces/ISessionStorage";
+
+export class SessionFileStorage implements ISessionStorage
+{
+	private smartAccountAddress: string;
+
+	constructor ( smartAccountAddress: string )
+	{
+		this.smartAccountAddress = smartAccountAddress.toLowerCase();
+	}
+	// This method reads data from the file and returns it in the JSON format
+	private async readDataFromFile ( type: "sessions" | "signers" ): Promise<any>
+	{
+		return new Promise( ( resolve ) =>
+		{
+			fs.readFile( this.getStorageFilePath( type ), "utf8", ( err, data ) =>
+			{
+				if ( err )
+				{
+					// Handle errors appropriately
+					resolve( undefined );
+				} else
+				{
+					if ( !data )
+					{
+						resolve( null )
+					} else
+					{
+						resolve( JSON.parse( data ) );
+
+					}
+					//   resolve(JSON.parse(data));
+				}
+			} );
+		} );
+	}
+
+	private getStorageFilePath ( type: "sessions" | "signers" ): string
+	{
+		return `./${ this.smartAccountAddress }_${ type }.json`;
+	}
+
+	private async writeDataToFile ( data: any, type: "sessions" | "signers" ): Promise<void>
+	{
+		return new Promise( ( resolve, reject ) =>
+		{
+			const filePath = this.getStorageFilePath( type );
+			fs.writeFile( filePath, JSON.stringify( data ), "utf8", ( err ) =>
+			{
+				if ( err )
+				{
+					// Handle errors appropriately
+					reject( err );
+				} else
+				{
+					resolve();
+				}
+			} );
+		} );
+	}
+
+	private validateSearchParam ( param: SessionSearchParam ): void
+	{
+		if ( param.sessionID )
+		{
+			return;
+		} else if ( !param.sessionID && param.sessionPublicKey && param.sessionValidationModule )
+		{
+			return;
+		} else
+		{
+			throw new Error( "Either pass sessionId or a combination of sessionPublicKey and sessionValidationModule address." );
+		}
+	}
+	// Session store is in the form of mekrleRoot and leafnodes, each object will have a root and an array of leafNodes.
+	private async getSessionStore (): Promise<any>
+	{
+		try
+		{
+			const data = await this.readDataFromFile( "sessions" );
+			return data || { merkleRoot: "", leafNodes: [] };
+		} catch ( error )
+		{
+			// Handle errors appropriately
+			throw error;
+		}
+	}
+
+	private async getSignerStore (): Promise<any>
+	{
+		try
+		{
+			const data = await this.readDataFromFile( "signers" );
+			return data || {};
+		} catch ( error )
+		{
+			// Handle errors appropriately
+			throw error;
+		}
+	}
+
+	private getStorageKey ( type: "sessions" | "signers" ): string
+	{
+		return `${ this.smartAccountAddress }_${ type }`;
+	}
+
+	private toLowercaseAddress ( address: string ): string
+	{
+		return address.toLowerCase();
+	}
+
+	async getSessionData ( param: SessionSearchParam ): Promise<SessionLeafNode>
+	{
+		// this.validateSearchParam(param);
```

Comment:
why is this commented out

Reply:
this is for validating the params, this gets used when managing multiple users session keys, have avoided given we are now creating leafs for one user only

Reply:
think about real world scenario how it can be used by multiple users and dapps. like I mentioned files should also be created automatically. cause people might try and use such storage client on their real apps. 
you can make sure search works uncomment it and leave the comment above it for it's purpose.